src/components/images/attributes.js

```diff
@@ -0,0 +1,158 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var cartesianConstants = require('../../plots/cartesian/constants');
+
+
+module.exports = {
+    _isLinkedToArray: true,
+
+    source: {
+        valType: 'string',
+        role: 'info',
+        description: [
+            'Specifies the URL of the image to be used.',
+            'The URL must be accessible from the domain where the',
+            'plot code is run, and can be either relative or absolute.'
+
+        ].join(' ')
+    },
+
+    layer: {
+        valType: 'enumerated',
+        values: ['below', 'above'],
+        dflt: 'above',
+        role: 'info',
+        description: [
+            'Specifies whether images are drawn below or above traces.',
+            'When `xref` and `yref` are both set to `paper`,',
+            'image is drawn below the entire plot area.'
+        ].join(' ')
+    },
+
+    width: {
```

Comment:
I'm not a fan of using `width` and `height` for relative dimensions like here.

Maybe we could use `domain.x` and `domain.y`, for example

```js
{
  domain: {
    x: [0, 0.5],
    y: [0, 0.5]
  }
}
```  

would be equivalent to 


```js
{
   x: 0,
   y: 0,
   width: 0.5,
   height: 0.5
}
```

cc @cldougl @chriddyp